### ISSUE TITLE
The performance loss of xputimer training model is nearly 20%

### ISSUE THREAD

**Issue by aluode99** [intelligent-machine-learning/dlrover#1546]
* environment information is as follows： 
  *  gpu: H100 *8 
   * pytorch: 25.03-py3
   * nccl: 2.21.5
   * cuda: 12.3
  
* During the xputimer demo test, the average time consumed by each Epoch within 1 hour is as follows (excluding the first abnormal data)：

| without xputimer   | with xputimer       | Loss rate| 
|------------|----------------|----------------------|
| 0.14475  |         0.152834   |  (0.152834-0.14475)/0.14475=5.58%|

 As shown in the figure below, after removing the first data 5.591107130050659, the average value of all time consumption is calculated

![Image](https://github.com/user-attachments/assets/2ef61009-5eae-4343-9fd7-ff20522ef3ec)

* magetron  gpt3 175b
   *  args 
      GPT_MODEL_ARGS=(
    --num-layers 12
    --hidden-size 1024
    --num-attention-heads 64
    --seq-length 2048
    --max-position-embeddings 2048
    --attention-backend auto # Can use (flash/fused/unfused/local)
     )
     TRAINING_ARGS=(
    --micro-batch-size 1
    --global-batch-size 32
    #--rampup-batch-size 16 16 5859375
    --train-iters 500000
    --weight-decay 0.1
    --adam-beta1 0.9
    --adam-beta2 0.95
    --init-method-std 0.006
    --clip-grad 1.0
    --fp16
    --lr 6.0e-5
    --lr-decay-style cosine
    --min-lr 6.0e-6
    --lr-warmup-fraction .001
    --lr-decay-iters 430000
    )
            
   * Train 10,000 times, remove the first abnormal data, and calculate the average time consumption as follows：
    
|  model                 | without xputimer   | with xputimer            | Loss rate| 
|-----------------|------------|----------------|----------------------|
| tp=1 pp=1 dp=8 | 177.47         |        219.46       | (219.46-177.47)/177.47=23.66%|
| tp=2 pp=1 dp=4  | 385.38      |        472.60       | (472.60-385.38)/472.60=18.46%|
| tp=2 pp=2 dp=2  | 424.03      |        507.45       | (507.45-424.03)/424.03=19.67%|

* According to the official documentation, xputimer brings very little performance loss：

<img width="839" alt="Image" src="https://github.com/user-attachments/assets/ef3f536c-5f6b-42a4-88fd-ef2e217d7ad8" />

* The loss of 3D scene is about 5%

<img width="1104" alt="Image" src="https://github.com/user-attachments/assets/ff0d1a77-389b-4fc3-9771-c3e8043bd2c9" />

During my test, there was a performance loss of nearly 20%. Is there something wrong with my settings?

**Comment by cos120** [comment]
Performance cost only happens on
1. your model has host launch bound or have lots of synchronizations.
2. For Megatron-LM, since we use `cudaEvent` with timing flag and dues to `CUDA_MAX_DEVICE_CONNECTIONS=1`,  the `cudaEvent` will break the overlapping of zero1 and backward of tp. `cudaEvent` with timing flag acts like kernel and affected by `CUDA_MAX_DEVICE_CONNECTIONS`, this will let kernels and **cudaEvnet** launch as ordered as the order of cpu launch kernel or call `cudaEventRecord`. We have developed new tracing on CUPTI, but it not release yet.
